cmd/client/main.go

```diff
@@ -31,18 +32,15 @@ func main() {
 		os.Exit(0)
 	}
 
-	if scanner.NoTools() {
-		log.Println("[Core] No scanner tools available")
-		os.Exit(0)
-	}
+	var pluginRegistry plugin.Register = loadPluginRegistry(flg.regFile)
 
-	scn, found := scanner.FromStr(flg.scanner)
-	if flg.scanner == "" || !found {
-		scn = scanner.Default
-		log.Printf("[Core] Scanner tool not specified or not found, default scanner tool %v is selected instead\n", scn.Name)
+	scn, found := pluginRegistry.FromStr(flg.scanner)
+	if !found {
+		scn = pluginRegistry.Default()
+		log.Printf("[Core] Unable to find scanner plugin with name '%s'; fallback to default scanner with name '%s'", flg.scanner, scn.Name)
 	}
 
-	cfg := &scanner.Config{Tool: scn, InDir: flg.inDir, ResultDir: "/tmp/pm/"}
```

Comment:
Is there a function that provides you a temp dir given by the OS?

Reply:
Yes, I used it to create the log file, I will adapt this line to use the same function

Reply:
This is fixed in 6ace911